Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM registry.access.redhat.com/rhscl/ruby-23-rhel7
+
+# sshd
+USER root
+RUN yum install -y --setopt=tsflags=nodocs openssh-server libicu-devel && \
+    yum clean all && /usr/sbin/sshd-keygen
+
+RUN mkdir /var/run/sshd
+#RUN sed -i 's/LogLevel INFO/LogLevel VERBOSE/' /etc/ssh/sshd_config
+RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+# gems
+#ENV GEM_HOME="/usr/local/lib/ruby/gems/2.1.0"
+RUN ["bash", "-c", "gem install --no-ri --no-rdoc bunny"]
+
+# git user
+RUN groupadd -r git &&\
+     useradd -r -s /bin/bash -g git git && \
+     mkdir --parent /home/git && \
+     chown -R git:git /home/git
+
+
+# gitlab-shell setup
+COPY . /home/git/gitlab-shell
+WORKDIR /home/git/gitlab-shell
+RUN ["bash", "-c", "./bin/install"]
+
+COPY authorized_keys /home/git/.ssh/authorized_keys
+COPY dummy-redis-cli.sh /usr/bin/redis-cli
```

Comment:
This is expected and is in fact how it works in SaaS production today.
Gitlab require redis.
Gitlab-shell uses a redis queue to notify Gitlab.
Gitlab-shell shells out to the redis cli to do this notification.
We don't require this as we don't have Gitlab.
A dummy-cli ensures Gitlab-shell doesn't cause an error.